crates/typst/src/export/pdf/color.rs

```diff
@@ -0,0 +1,443 @@
+                .range([0.0, 1.0, 0.0, 1.0, 0.0, 1.0])
+                .filter(Filter::FlateDecode);
+        }
+
+        // Write the sRGB color space
+        if let Some(srgb) = self.srgb {
+            let profile = srgb_icc();
+            writer
+                .icc_profile(srgb, &profile)
+                .n(3)
+                .range([0.0, 1.0, 0.0, 1.0, 0.0, 1.0])
+                .alternate()
+                .srgb();
+        }
+
+        // Write the gray color space
+        if let Some(gray) = self.d65_gray {
+            let profile = gray_icc();
+            writer
+                .icc_profile(gray, &profile)
+                .n(1)
+                .range([0.0, 1.0])
+                .alternate()
+                .d65_gray();
+        }
+    }
+}
+
+// The ICC profiles
+const SRGB_ICC: &[u8] = include_bytes!("../../../icc/sRGB-v4.icc");
+const GRAY_ICC: &[u8] = include_bytes!("../../../icc/sGrey-v4.icc");
+
+///The PostScript functions for color spaces
+const OKLAB_SOURCE: &str = include_str!("../../../post-script/oklab.ps");
+const HSL_SOURCE: &str = include_str!("../../../post-script/hsl.ps");
+const HSV_SOURCE: &str = include_str!("../../../post-script/hsv.ps");
+
+// The name of the color spaces
+pub const SRGB: Name<'static> = Name(b"srgb");
+pub const D65_GRAY: Name<'static> = Name(b"d65gray");
+pub const OKLAB: Name<'static> = Name(b"oklab");
+pub const HSV: Name<'static> = Name(b"hsv");
+pub const HSL: Name<'static> = Name(b"hsl");
+pub const LINEAR_SRGB: Name<'static> = Name(b"linearrgb");
+
+// The name of the color components
+const OKLAB_L: Name<'static> = Name(b"L");
+const OKLAB_A: Name<'static> = Name(b"A");
+const OKLAB_B: Name<'static> = Name(b"B");
+const HSV_H: Name<'static> = Name(b"H");
+const HSV_S: Name<'static> = Name(b"S");
+const HSV_V: Name<'static> = Name(b"V");
+const HSL_H: Name<'static> = Name(b"H");
+const HSL_S: Name<'static> = Name(b"S");
+const HSL_L: Name<'static> = Name(b"L");
+
+/// Deflated sRGB ICC profile
+#[comemo::memoize]
+fn srgb_icc() -> Arc<Vec<u8>> {
+    Arc::new(deflate(SRGB_ICC))
+}
+
+/// Deflated gray ICC profile
+#[comemo::memoize]
+fn gray_icc() -> Arc<Vec<u8>> {
+    Arc::new(deflate(GRAY_ICC))
+}
+
+/// Deflated Oklab PostScript function
+#[comemo::memoize]
+fn oklab_function() -> Arc<Vec<u8>> {
+    let code = unscanny(OKLAB_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// Deflated HSV PostScript function
+#[comemo::memoize]
+fn hsv_function() -> Arc<Vec<u8>> {
+    let code = unscanny(HSV_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// Deflated HSL PostScript function
+#[comemo::memoize]
+fn hsl_function() -> Arc<Vec<u8>> {
+    let code = unscanny(HSL_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// This function removes comments, line spaces and carriage returns from a
+/// PostScript program. This is necessary to optimize the size of the PDF file.
+fn unscanny(source: &str) -> String {
```

Comment:
I think this is a misunderstanding. I didn't mean to rename to this to `unscanny`, but to use the `unscanny` crate (which is already a dep and used in the parser) to write this in a more understandable way:

```rust
/// This function removes comments, line spaces and carriage returns from a
/// PostScript program. This is necessary to optimize the size of the PDF file.
fn minify(source: &str) -> String {
    let mut buf = String::with_capacity(source.len());
    let mut s = unscanny::Scanner::new(source);
    while let Some(c) = s.eat() {
        match c {
            '%' => {
                s.eat_until('\n');
            }
            c if c.is_whitespace() => {
                s.eat_whitespace();
                if buf.ends_with(|c: char| !c.is_whitespace()) {
                    buf.push(' ');
                }
            }
            _ => buf.push(c),
        }
    }
    buf
}
```

Reply:
Ah! I didn't get that 😂